processing/src/main/java/io/druid/query/DefaultQueryMetrics.java

```diff
@@ -66,15 +65,6 @@ public void queryType(QueryType query)
   }
 
   @Override
-  public void interval(QueryType query)
```

Comment:
@jihoonson examples of what?

IMO `interval` is not a useful metric dimension. `duration` is a much more useful one. Additionally, `interval` has high cardinality, and consumes quite a lot of space in metric storing systems (i. e. Druid itself). This effect is multiplied, if interval is concatenated with data source in a dimension.

So we need to keep emit `interval` dimension for backwards compatibility for metrics of old query types, but for new query types, I would not do that. `dataSourcesAndIntervals()` Javadoc should tell "emit data source and interval information in whatever form you want", but emit just `dataSource` for the new join query type.